controllers/reconcilers/kserve_metrics_dashboard_reconciler.go

```diff
@@ -0,0 +1,223 @@
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// func (r *KserveMetricsDashboardReconciler) createDesiredResource(log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+func (r *KserveMetricsDashboardReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+
+	// resolve SR
+	isvcRuntime := isvc.Spec.Predictor.Model.Runtime
+	runtime := &kservev1alpha1.ServingRuntime{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: *isvcRuntime, Namespace: isvc.Namespace}, runtime); err != nil {
+		log.Error(err, "Could not determine servingruntime for isvc")
+	}
+
+	servingRuntimeImage := runtime.Spec.Containers[0].Image
+	re := regexp.MustCompile(`/([^/@]+)[@:]`)
+	findImageName := re.FindStringSubmatch(servingRuntimeImage)
+	servingRuntime := findImageName[1]
+
+	switch servingRuntime {
+	case "openvino_model_server":
+		if ovmsData == nil {
+			data, err := os.ReadFile("ovms-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+			}
+			err = json.Unmarshal(data, &ovmsData)
+			if err != nil {
+				log.Error(err, "Error unmarshalling JSON:", err)
+			}
+		}
+	case "text-generation-inference":
+		if tgisData == nil {
+			data, err := os.ReadFile("tgis-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+			}
+			err = json.Unmarshal(data, &tgisData)
+			if err != nil {
+				log.Error(err, "Error unmarshalling JSON:", err)
+			}
+		}
+	}
+
+	var configMapData MetricsDashboardConfigMapData
+	//TODO: move read file logic to switch and only read if global variable is nil
+	data, err := os.ReadFile("ovms-metrics.json")
+	if err != nil {
+		log.Error(err, "Unable to load metrics dashboard template file")
+	}
+
+	stringData := string(data)
+	stringDatawithNS := strings.Replace(stringData, "${namespace}", isvc.Namespace, -1)
+	stringDataComplete := strings.Replace(stringDatawithNS, "${model_name}", isvc.Name, -1)
+	data = []byte(stringDataComplete)
+	err = json.Unmarshal(data, &configMapData)
+	if err != nil {
+		log.Error(err, "Unable to load metrics dashboard templates")
+	}
+	jsonData, err := json.MarshalIndent(configMapData, "", " ")
+	if err != nil {
+		log.Error(err, "Unable to marshal data for metrics dashboard configmap")
+	}
+	log.V(1).Info("jsondata", "value", string(jsonData))
+	// Create ConfigMap object
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      isvc.Name + "-metrics-dashboard",
+			Namespace: isvc.Namespace,
+		},
+		Data: map[string]string{
+			"Data": string(jsonData), //TODO: multiple runtimes support
+		},
+	}
```

Comment:
might be good to separate the string replacement logic into a function for readability. Also since this is the final processing step of the data before we create the configmap, we can pass the function the bytearray we have in `data` and have it return the string that we will use in the configmap. 
```suggestion
	finaldata := substituteVariablesInQueries(data, isvc.Namespace, isvc.Name)
	// Create ConfigMap object
	configMap := &corev1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
			Name:      isvc.Name + "-metrics-dashboard",
			Namespace: isvc.Namespace,
		},
		Data: map[string]string{
			"metrics": finaldata,
		},
	}
```